core/genesis.go

```diff
@@ -596,28 +596,40 @@ func DefaultHoleskyGenesisBlock() *Genesis {
 }
 
 // DeveloperGenesisBlock returns the 'geth --dev' genesis block.
-func DeveloperGenesisBlock(gasLimit uint64, faucet common.Address) *Genesis {
+func DeveloperGenesisBlock(gasLimit uint64, faucet common.Address, genesisAlloc *GenesisAlloc) *Genesis {
 	// Override the default period to the user requested one
 	config := *params.AllDevChainProtocolChanges
+	alloc := map[common.Address]GenesisAccount{
```

Comment:
Hm, actually -- I'm having second thoughts here. If the user wants to provide an alloc, why should geth interfere with that? 
I think we should _either_ 
- use the "sane default" == a faucet + 1-wei precompiles, OR
- use the provided alloc. 

Doing a mishmash of both just makes for a strange UX.

Reply:
In my usage scenario, I need to test a contract, so I write the bytecode of that contract directly into genesis.json, no need to redeploy the contract again after `geth --dev` started.

Reply:
Hm,one more thing to consider, of course, is that the actual `faucet` address, `eth.accounts[0]` is ephemeral, so it's not known to the user before the process starts. 
I guess we can let it be as is, and ignore my comment above